docs/container-set-template.md

```diff
@@ -116,3 +116,54 @@ Example B: Lopsided requests, e.g. `a -> b` where `a` is cheap and `b` is expens
 Can you see the problem here? `a` only has small requests, but the container set will use the  total of all requests. So it's as if you're using all that GPU for 10h. This will be expensive.
 
 Solution: do not use container set when you have lopsided requests.
+
+## `retryStrategy` usage
+
+> v3.3 and after
+
+You can set a `retryStrategy` to apply to all containers of a container set.
+
+This currently works differently from [retries](retries.md) for other template types.
+You can only set the `duration` between each retry and the total number of `retries`.
+
+See an example below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: containerset-with-retrystrategy
+  annotations:
+    workflows.argoproj.io/description: |
+      This workflow creates a container set with a retryStrategy.
+spec:
+  entrypoint: containerset-retrystrategy-example
+  templates:
+    - name: containerset-retrystrategy-example
+      containerSet:
+        retryStrategy:
+          retries: "10" # if fails, retry at most ten times
+          duration: 30s # retry for at most 30s
+        containers:
+          # this container completes successfully, so it won't be retried.
+          - name: success
+            image: python:alpine3.6
+            command:
+              - python
+              - -c
+            args:
+              - |
+                print("hi")
+          # if fails, it will retry at most ten times.    
+          - name: fail-retry
+            image: python:alpine3.6
+            command: ["python", -c]
+            # fail with a 66% probability
+            args: ["import random; import sys; exit_code = random.choice([0, 1, 1]); sys.exit(exit_code)"]
+```
+
+!!! Note
+    A container set will not be retried if a container's `command` cannot be located.
+    As it will fail each time, the retry logic is short-circuited.
+    Since it is only a process-level retry, so won't create new node.
+    You can confirm whether a retry has been performed by checking the container's log.
```

Comment:
```suggestion
!!! Note Template-level `retryStrategy` vs Container Set `retryStrategy`
    `containerSet.retryStrategy` works differently from [template-level retries](retries.md):
    1. Your `command` will be re-ran by the Executor inside the same container if it fails.
        - As no new containers are created, the nodes in the UI remain the same, and the retried logs are appended to original container's logs. For example, your container logs may look like:
        ```
        time="2024-03-29T06:40:25 UTC" level=info msg="capturing logs" argo=true
        intentional failure
        time="2024-03-29T06:40:25 UTC" level=debug msg="ignore signal child exited" argo=true
        time="2024-03-29T06:40:26 UTC" level=info msg="capturing logs" argo=true
        time="2024-03-29T06:40:26 UTC" level=debug msg="ignore signal urgent I/O condition" argo=true
        intentional failure
        time="2024-03-29T06:40:26 UTC" level=debug msg="ignore signal child exited" argo=true
        time="2024-03-29T06:40:26 UTC" level=debug msg="forwarding signal terminated" argo=true
        time="2024-03-29T06:40:27 UTC" level=info msg="sub-process exited" argo=true error="<nil>"
        time="2024-03-29T06:40:27 UTC" level=info msg="not saving outputs - not main container" argo=true
        Error: exit status 1
        ```
    1. If a container's `command` cannot be located, it will not be retried.
        - As it will fail each time, the retry logic is short-circuited.
```

- add a title to the note as there's a common subject now between these
- move the "different from template-level retries" part into this note, including the link to `retries.md`
- be more specific about the "nodes" in the UI / `status` (many users don't know what "nodes" mean as it's part of the `status` rather than user-facing per se) as well as what happens with the logs
  - add an example of the logs for optimal clarity